pkg/steps/template.go

```diff
@@ -561,11 +562,16 @@ func waitForPodCompletionOrTimeout(ctx context.Context, podClient PodClient, nam
 				log.Printf("warning: failed to get pod %s: %v", name, err)
 				continue
 			}
-			if !podHasStarted(pod) && time.Since(pod.CreationTimestamp.Time) > podStartTimeout {
-				message := fmt.Sprintf("pod didn't start running within %s: %s\n%s", podStartTimeout, getReasonsForUnreadyContainers(pod), getEventsForPod(ctx, pod, podClient))
-				log.Print(message)
-				notifier.Complete(name)
-				return pod, errors.New(message)
+
+			if !podSeenRunning {
+				if podHasStarted(pod) {
```

Comment:
This returns `false` for pods that finished, making us incorrectly mark them as timeout:
https://github.com/openshift/ci-tools/blob/b5d126c9ade3189ca9ee7d5621a75055cc930b0a/pkg/steps/template.go#L593-L604